ruby-service.sql

```diff
@@ -0,0 +1,79 @@
+-- Drop previous versions of the tables if they they exist, in reverse order of foreign keys.
+
+DROP TABLE IF EXISTS Events;
+DROP TABLE IF EXISTS Trailer;
+DROP TABLE IF EXISTS Item;
+Drop TABLE IF EXISTS TrailerUser;
+DROP TABLE IF EXISTS TrailerItem;
+DROP TABLE IF EXISTS Users;
+
+-- Create the schema.
+CREATE TABLE Users (
+  username varchar(25) PRIMARY KEY NOT NULL,
+  password varchar(50) NOT NULL,
+	type varchar(10) NOT NULL
+	);
+
+CREATE TABLE Events (
+	ID integer PRIMARY KEY, 
+	name varchar(50),
+  time timestamp,
+  description text
+	);
+
+CREATE TABLE Trailer (
+  ID integer PRIMARY KEY, 
+	name varchar(25)
+	);
+
+CREATE TABLE Item (
+  ID integer PRIMARY KEY, 
+  name varchar(25),
+  quantity integer,
+  notificationlevel integer,
+  increment integer
+  );
+
+CREATE TABLE TrailerUser (
```

Comment:
Is this going to be a one to one relationship (only one login per trailer)?  If so, we could just add a userID field to the Trailer table

Reply:
Ok, looking at the domain model I see where you're getting this.  I do think, though, that it would be possible to put a userID per trailer and just assume each admin can see all trailers (so there is no many admin to many trailers relationship that has to be specified). This way would work too though

Reply:
I considered that, and it works both ways. However, having a separate user-trailer access table simplifies the query logic, so we don't need an if statement based on whether the user is an admin. It's slightly more flexible for small numbers of users and trailers